File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
    let answer = [];
    operations.forEach((operation)=>{
        const opArr = operation.split(' ');
        if(opArr[0]==='I') {
            answer.push(+opArr[1]);
        }
        if(opArr[0]==='D') {
            if(opArr[1][0]==='-'){
                answer.shift();
            }else{
                answer.pop();
            }
        }
        answer.sort((a, b)=>a-b);
    })
    return answer.length === 0 ? [0, 0] : [answer[answer.length-1], answer[0]];
}

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
        
        for(String operation : operations) {
            String[] parts = operation.split(" ");
            String op = parts[0];
            Integer value = Integer.parseInt(parts[1]);
            
            if (op.equals("I")) {
                minHeap.offer(value);
                maxHeap.offer(value);
            } else if (op.equals("D")) {
                if(minHeap.isEmpty()) continue;
                
                if (value > 0) {
                    int max = maxHeap.poll();
                    minHeap.remove(max);
                } else if (value < 0) { 
                    int mix = minHeap.poll();
                    maxHeap.remove(mix);
                }
            }
        }
        
        
        if(minHeap.isEmpty()) {
            return new int[]{0, 0};    
        }
        
        return new int[]{maxHeap.peek(), minHeap.peek()};
    }
}